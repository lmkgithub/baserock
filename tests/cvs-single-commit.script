#!/bin/sh
#
# Tests converting a simple CVS repository to git.
#
# Copyright (C) 2012  Codethink Limited
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; version 2 of the License.
# 
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
# 
# You should have received a copy of the GNU General Public License along
# with this program; if not, write to the Free Software Foundation, Inc.,
# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.


set -e


logfile="$DATADIR/cvs-test-repo.log"
workdir="$DATADIR/work-dir"

# ./lorry --pull-only --log="$logfile" --working-area="$workdir" \
#   "$DATADIR/cvs-test-repo.lorry" # > /dev/null 2> /dev/null
# 
# #export GIT_DIR="$workdir/cvs-test-repo/git/.git"
# cd "$workdir/cvs-test-repo/git/"
# 
# # list the branches
# git show-ref | cut -d' ' -f2
# 
# # cat the test file
# git cat-file blob trunk:test.txt
# 
# # check out the trunk branch
# git checkout --quiet trunk
# 
# # list the commit messages
# git log --pretty='%s'
