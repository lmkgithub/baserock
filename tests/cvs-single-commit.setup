#!/bin/sh
#
# Creates a CVS repository with a single file and a single commit.
#
# Copyright (C) 2012  Codethink Limited
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; version 2 of the License.
# 
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
# 
# You should have received a copy of the GNU General Public License along
# with this program; if not, write to the Free Software Foundation, Inc.,
# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.


set -e

# clean up the data directory
test -d "$DATADIR" && rm -rf "$DATADIR"/*

# create the repository
repo="$DATADIR/cvs-test-repo"
export CVSROOT="$repo"
export CVS_RSH=
cvs init

# create a local working copy
workingcopy="$DATADIR/cvs-test-checkout"
mkdir "$workingcopy"
cd "$workingcopy"
cvs checkout CVSROOT
cd "$workingcopy/CVSROOT"

# add the test file
echo "first line" > test.txt
cvs -Q add test.txt

# make a commit
cvs -Q commit -m "first commit"

# delete the working copy
cd "$DATADIR"
rm -r "$workingcopy"

# create the .lorry file for the repository
cat <<EOF > $DATADIR/cvs-test-repo.lorry
{
  "cvs-test-repo": {
    "type": "cvs",
    "url": "$repo",
    "module": "CVSROOT"
  }
}
EOF


# create the working directory
mkdir "$DATADIR/work-dir"
