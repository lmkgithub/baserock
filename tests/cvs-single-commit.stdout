branches:
refs/heads/master
refs/heads/origin
test file:
first line
commit messages:
first commit
initial checkin
